From: =?utf-8?q?Christian_G=C3=B6ttsche?= <cgzones@googlemail.com>
Date: Mon, 2 Feb 2026 14:53:20 +0100
Subject: Fix matchpathcon_filespec_add() condition for alpha

Correctly handle the case when unsigned long is 8 bytes wide, but ino_t
only 4 bytes, like on alpha. Use __BITS_PER_LONG already included by
selinux.h.
---
 include/selinux/selinux.h | 3 ++-
 src/matchpathcon.c        | 3 ++-
 2 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/include/selinux/selinux.h b/include/selinux/selinux.h
index de56b72..c63d174 100644
--- a/include/selinux/selinux.h
+++ b/include/selinux/selinux.h
@@ -509,7 +509,8 @@ extern int matchpathcon_index(const char *path, mode_t mode, char **con);
    use the latter of the two specifications based on their order in the 
    file contexts configuration.  Return the used specification index. */
 #if defined(_FILE_OFFSET_BITS) && _FILE_OFFSET_BITS == 64 && \
-	defined(__INO64_T_TYPE) && !defined(__INO_T_MATCHES_INO64_T)
+	defined(__INO64_T_TYPE) && !defined(__INO_T_MATCHES_INO64_T) && \
+	__BITS_PER_LONG < 64
 #define matchpathcon_filespec_add matchpathcon_filespec_add64
 #endif
 extern int matchpathcon_filespec_add(ino_t ino, int specind, const char *file);
diff --git a/src/matchpathcon.c b/src/matchpathcon.c
index 2623e5d..acc7a9a 100644
--- a/src/matchpathcon.c
+++ b/src/matchpathcon.c
@@ -281,7 +281,8 @@ oom:
 }
 
 #if (defined(_FILE_OFFSET_BITS) && _FILE_OFFSET_BITS == 64) && \
-	defined(__INO64_T_TYPE) && !defined(__INO_T_MATCHES_INO64_T)
+	defined(__INO64_T_TYPE) && !defined(__INO_T_MATCHES_INO64_T) && \
+	__BITS_PER_LONG < 64
 /* alias defined in the public header but we undefine it here */
 #undef matchpathcon_filespec_add
 
